atomicapp/providers/kubernetes.py

```diff
@@ -38,227 +44,243 @@ class KubernetesProvider(Provider):
+            '''
+            Config_file:
+                If a config_file has been provided, use the configuration
+                from the file and load the associated generated file.
+                If a config_file exists (--provider-config) use that.
 
-    def _find_kubectl(self, prefix=""):
-        """Determine the path to the kubectl program on the host.
-        1) Check the config for a provider_cli in the general section
-           remember to add /host prefix
-        2) Search /usr/bin:/usr/local/bin
+            Params:
+                If any provider specific parameters have been provided,
+                load the configuration through the answers.conf file
 
-        Use the first valid value found
-        """
+            .kube/config doesn't exist:
+                If the configuration file does not exist, generate one using
+                the default localhost:8080/api end-point.
+
+            Default configuration:
+                If no paramaters have been provided, as well as no --provider-config,
+                use the default .kube/config location and load the associated configuration.
+            '''
 
-        if self.dryrun:
-            # Testing env does not have kubectl in it
-            return "/usr/bin/kubectl"
+            config_location = os.path.join(Utils.getRoot(), Utils.getUserHome().strip('/'), '.kube/config')
 
-        test_paths = ['/usr/bin/kubectl', '/usr/local/bin/kubectl']
-        if self.config.get("provider_cli"):
-            logger.info("caller gave provider_cli: " + self.config.get("provider_cli"))
-            test_paths.insert(0, self.config.get("provider_cli"))
+            if self.config_file:
+                self.api = Client(KubeConfig.from_file(self.config_file), "kubernetes")
+            elif self._check_required_params():
+                self.api = Client(self._from_required_params(), "kubernetes")
+            elif not os.path.isfile(config_location):
+                self.config["provider-api"] = self.provider_api
+                self.api = Client(self._from_required_params(), "kubernetes")
+            else:
+                self.api = Client(KubeConfig.from_file(config_location), "kubernetes")
+
+            # Check if the namespace that the app is being deployed to is available
+            self._check_namespaces()
+
+    def _check_required_params(self):
+        '''
+        This checks to see if any parameters associated to the Kubernetes provider are passed
+        '''
+
+        # Initialize the values
+        result = {PROVIDER_API_KEY: self.provider_api,
+                  PROVIDER_AUTH_KEY: self.provider_auth,
+                  PROVIDER_TLS_VERIFY_KEY: self.provider_tls_verify,
+                  PROVIDER_CA_KEY: self.provider_ca}
+
+        # Get values from the loaded answers.conf / passed CLI params
+        for k in result.keys():
+            result[k] = self.config.get(k)
+
+        if bool([a for a in result.values() if a is None]):
```

Comment:
I just don't like it when the alternative (much easier to read) version right there for the taking :) 

I was only making a suggestion so we can leave it like it is, but do you agree that one is easier to read than the other?